src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java

```diff
@@ -421,13 +430,34 @@ private ScheduleRunConfiguration getScheduleRunConfiguration(Boolean isRunUnmete
         }
         return suites;
     }
+    
+    private Map<String, FilePath> getTests(AWSDeviceFarm adf, ScheduleRunResult run, Map<String, FilePath> suites) throws IOException, InterruptedException {
+        Map<String, FilePath> tests = new HashMap<String, FilePath>();
+        String runArn = run.getRun().getArn();
+        String components[] = runArn.split(":");
+        // constructing suite ARN for each job using the run ARN
+        components[5] = "suite";
+        for(String suiteArn:suites.keySet()) {
+            components[6] = suiteArn;
+            String fullsuiteArn = StringUtils.join(components,":");
+            ListTestsResult result = adf.listTests(fullsuiteArn);
+            for (Test test : result.getTests()) {
+                String arn = test.getArn().split(":")[6];
+                tests.put(arn, new FilePath(suites.get(suiteArn), test.getName()));
+                tests.get(arn).mkdirs();
+            }
+        }
+        return tests;
+    }
 
     private Map<String, FilePath> getJobs(AWSDeviceFarm adf, ScheduleRunResult run, FilePath resultsDir) throws IOException, InterruptedException {
         Map<String, FilePath> jobs = new HashMap<String, FilePath>();
         ListJobsResult result = adf.listJobs(run.getRun().getArn());
         for (Job job : result.getJobs()) {
             String arn = job.getArn().split(":")[6];
-            jobs.put(arn, new FilePath(resultsDir, job.getName()));
+            String jobId = arn.substring(arn.lastIndexOf("/") + 1);
+            // Two jobs can have same name
+            jobs.put(arn, new FilePath(resultsDir, job.getName() + "-" + jobId));
```

Comment:
I can't remember...does the Job contain a reference to the device in the object? If so, i'd prefer you use a combination of the name + device version instead of the job ID, as the auto-increment number is not the most user friendly or helpful value.